tools/incubator.js

```diff
@@ -85,6 +98,25 @@ const Incubator = new Lang.Class({
         return basin_metadata;
     },
 
+    _import_video: function (asset) {
+        let basin_metadata = new Map();
+
+        let file = this._download_file(asset['uri']);
+        let info = file.query_info('*', Gio.FileQueryInfoFlags.NONE, null);
+
+        basin_metadata['source'] = file.get_path();
```

Comment:
Just realized another pre-existing defect that I didn't spot last time: you're setting these keys as properties on the `Map` object, and not as keys in the map's hashtable. To access the hashtable, you have to use `get()`, `set()`, and `has()`.

Reply:
ouch, I feel very silly. Can I fix this all-in-once in a posterior commit?

Reply:
:+1: